plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -87,13 +88,35 @@ public Set<HiveColumnStatisticType> getSupportedColumnStatistics(Type type)
 
     public PartitionStatistics getTableStatistics(String databaseName, String tableName)
     {
-        return delegate.getTableStatistics(getExistingTable(databaseName, tableName));
+        return getTableStatistics(databaseName, tableName, Optional.empty());
+    }
+
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName, Optional<Set<String>> columns)
+    {
+        Table table = getExistingTable(databaseName, tableName);
+        if (columns.isPresent()) {
+            Set<String> requestedColumnNames = columns.get();
+            List<Column> requestedColumns = table.getDataColumns().stream().filter(column -> requestedColumnNames.contains(column.getName())).collect(toImmutableList());
```

Comment:
Columns could be empty set? — should we do anything with it? 

nit. Maybe apply different formatting for streaming?
```suggestion
            List<Column> requestedColumns = table.getDataColumns().stream()
                    .filter(column -> requestedColumnNames.contains(column.getName()))
                    .collect(toImmutableList());
```

Reply:
> Columns could be empty set? — should we do anything with it?

Theoretically, we could get only the table row count without the column stats (`count(*)` query). This is where the empty set can be used (it is not at the moment)

Reply:
formatting applied